src/index.tsx

```diff
@@ -1,5 +1,53 @@
 import * as React from 'react';
 import * as ReactDom from 'react-dom';
-import { App } from './App';
+import { connect, Provider } from 'react-redux';
+import { applyMiddleware, compose, createStore } from 'redux';
+import { createEpicMiddleware } from 'redux-observable';
+import { persistReducer, persistStore } from 'redux-persist';
+import { PersistGate } from 'redux-persist/integration/react';
+import storage from 'redux-persist/lib/storage';
+import App from './App';
+import { fetchMoviesAction, showDetailsAction } from './state/actions';
+import epics from './state/epics';
+import reducer, { State } from './state/reducers';
 
-ReactDom.render(<App />, document.getElementById('app'));
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
+const epicMiddleware = createEpicMiddleware();
+
+const persistConfig = {
+    key: 'root',
+    storage,
+};
+
+const persistedReducer = persistReducer(persistConfig, reducer);
+
+const store = createStore(persistedReducer,
```

Comment:
It will be much readable if extract store creating into a separate function. Connected App can also be located in its own directory.

Reply:
I have moved store creation into separate file

Reply:
App moved too